pkg/rbd/nodeserver.go

```diff
@@ -47,14 +48,25 @@ type NodeServer struct {
 // path
 func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
 	targetPath := req.GetTargetPath()
-	targetPathMutex.LockKey(targetPath)
-	disableInUseChecks := false
+	if targetPath == "" {
+		return nil, status.Error(codes.InvalidArgument, "Empty target path in request")
+	}
 
+	if req.GetVolumeCapability() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Empty volume capability in request")
+	}
+
+	if req.GetVolumeId() == "" {
+		return nil, status.Error(codes.InvalidArgument, "Empty volume ID in request")
+	}
+
```

Comment:
s/Empty/s in above if's..

Reply:
**NOTE**: Retaining one comment on the capitalization of first letter of messages in status.Error for ease of discussion.

As [pointed](https://github.com/ceph/ceph-csi/pull/312#discussion_r282756648) out by @Madhu-1 error strings need to start with a lower case letter in golang. This is also enforced by the linter.

These errors seem to be RPC error messages, **should they or do they follow the same convention** is something to consider before fixing it.

I assume we open a new issue against the code, to fix this across rather than address part of them here and rest later. IOW, I intend to leave added status.Errors in the current form than correcting them to adhere to the first letter being lower case.

@humblec or @Madhu-1 if you feel strongly that this should be corrected in this PR, for messages added with this PR, let me know.